core/src/main/scala/com/spotify/featran/transformers/OneHotEncoder.scala

```diff
@@ -53,39 +62,62 @@ private class OneHotEncoder(name: String) extends BaseHotEncoder[String](name) {
           fb.add(name + '_' + k, 1.0)
           fb.skip(math.max(0, c.size - v - 1))
         case None =>
-          fb.skip(c.size)
+          addMissingItem(c, fb)
           fb.reject(this, FeatureRejection.Unseen(Set(k)))
       }
-      case None => fb.skip(c.size)
+      case None => addMissingItem(c, fb)
     }
   }
 }
 
-private abstract class BaseHotEncoder[A](name: String)
+object MissingValue {
+  val missingValueToken = "__unknown__"
+}
+
+private abstract class BaseHotEncoder[A](name: String, encodeMissingValue: Boolean = false)
   extends Transformer[A, Set[String], SortedMap[String, Int]](name) {
 
+  val missingValueToken = MissingValue.missingValueToken
+
   def prepare(a: A): Set[String]
 
+  def addMissingItem(c: SortedMap[String, Int],
+                     fb: FeatureBuilder[_]): Unit = encodeMissingValue match {
+    case true =>
+      val v = c.get(missingValueToken).get // manually added so will exist
+      fb.skip(v)
+      fb.add(name + '_' + missingValueToken, 1.0)
+      fb.skip(math.max(0, c.size - v - 1))
+    case false => fb.skip(c.size)
+  }
+
   private def present(reduction: Set[String]): SortedMap[String, Int] = {
     val b = SortedMap.newBuilder[String, Int]
     var i = 0
-    val array = reduction.toArray
+    val array = encodeMissingValue match {
```

Comment:
if/else
Also array append is O(n), so `(reduction + missingValueToken).toArray` might be cheaper.